### PR TITLE
added experiences

### PR DESCRIPTION
**Pull request by hopeross**

### CODE REVIEW

File: Models/Project/Experiences.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    [Required]
+    public int UserId { get; set; }
+    public UserV1 User { get; set; }
+
+    [Required]
+    public string ExperienceTitle { get; set; }
+    public string ExperienceNotes { get; set; }
+    public bool Completed { get; set; }
+    public string ExperienceUserLocation { get; set; }
+
```
Comment (kendall-mchenry) [inline Models/Project/Experiences.cs:20]: Do we want to add a default CreatedOn date?

File: Repositories/ExperiencesRepository.cs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    private readonly MyPlacesDbContext _context;
+
+    public ExperiencesRepository(MyPlacesDbContext context)
+    {
+        _context = context;
+    }
+
+    public Experiences CreateExperience(Experiences newExperience)
+    {
+        _context.Experiences.Add(newExperience);
```
Comment (kendall-mchenry) [inline Repositories/ExperiencesRepository.cs:17]: We had this in MyPlaces. Do we want something like it here as well?
newExperience.CreatedOn = DateTime.Now.ToString();

File: Repositories/ExperiencesRepository.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        var experienceList = _context.Experiences.Where(e => e.UserId == userId).ToList();
+        return experienceList;
+    }
+
+    public Experiences GetExperiencesById(int experienceId)
+    {
+        return _context.Experiences.SingleOrDefault(e => e.ExperienceId == experienceId);
+    }
+
+    public Experiences GetExperiencesByUserIdGoogleId(int userId, string googlePlaceId)
```
Comment (kendall-mchenry) [inline Repositories/ExperiencesRepository.cs:33]: I don't think we'll need this but might as well leave it for now.

File: Repositories/ExperiencesRepository.cs [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    {
+        var experienceList = this.GetAllExperiencesByUserId(userId);
+
+        var foundPlace = experienceList.FirstOrDefault(e => e.FirstGooglePlaceId == googlePlaceId);
```
Comment (kendall-mchenry) [inline Repositories/ExperiencesRepository.cs]: Should this be `var = foundExperience`?

File: Repositories/ExperiencesRepository.cs [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+        return foundPlace;
+    }
```
Comment (kendall-mchenry) [inline Repositories/ExperiencesRepository.cs:40]: Same -- `foundExperience`

File: Models/Project/UserV1.cs [hunk 6]
```diff
@@ -32,5 +32,8 @@ public class UserV1
     [JsonIgnore]
     public List<MyPlace>? MyPlaces { get; set; }
 
+    [JsonIgnore]
+    public List<Experiences> Experiences { get; set; }
```
Comment (kendall-mchenry) [inline Models/Project/UserV1.cs:36]: Might need the ? after `List<Experiences>`

File: Migrations/MyPlacesDbContext.cs [hunk 7]
```diff
@@ -40,31 +40,45 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
...
+            entity.HasOne(e => e.User)
+            .WithMany(u => u.Experiences)
+            .HasForeignKey(e => e.UserId).IsRequired();
+            entity.Property(e => e.ExperienceTitle);
+            entity.Property(e => e.ExperienceNotes);
+            entity.Property(e => e.Completed);
+            entity.Property(e => e.ExperienceUserLocation);
+            entity.Property(e => e.FirstGooglePlaceId).IsRequired();
+            entity.Property(e => e.SecondGooglePlaceId);
+            entity.Property(e => e.ThirdGooglePlaceId);
```
Comment (kendall-mchenry) [inline Migrations/MyPlacesDbContext.cs:82]: Add CreatedOn property here if we want

File: Controllers/Project/ExperiencesController.cs [hunk 8]
```diff
@@ -0,0 +1,177 @@
...
+        if (HttpContext.User == null)
+        {
+            return Unauthorized();
+        }
+
+        var userIdClaim = HttpContext.User.Claims.FirstOrDefault(c => c.Type == "UserId");
+        var userId = Int32.Parse(userIdClaim.Value);
+        newExperience.UserId = userId;
+
+        if (userId == null)
```
Comment (kendall-mchenry) [inline Controllers/Project/ExperiencesController.cs:84]: I think this one if statement block can be removed

**Review by kendall-mchenry — APPROVED** [review]
Just a few small comments, but otherwise looks good to go!

File: Controllers/Project/ExperiencesController.cs (hunk 8 above)
Comment (hopeross) [reply Controllers/Project/ExperiencesController.cs:84]: No we need it you have to be signed in to create an experience

File: Models/Project/Experiences.cs (hunk 1 above)
Comment (hopeross) [reply Models/Project/Experiences.cs:20]: Are we actually using it anywhere?

File: Repositories/ExperiencesRepository.cs (hunk 4 above)
Comment (hopeross) [reply Repositories/ExperiencesRepository.cs]: fixed

File: Repositories/ExperiencesRepository.cs (hunk 5 above)
Comment (hopeross) [reply Repositories/ExperiencesRepository.cs:40]: eh it would have still worked since it used the same name but fixed

**Pull request merged**